### PR TITLE
chore: Enable codeql action

### PR DESCRIPTION
**Pull request by naveensrinivasan**
This action runs GitHub's industry-leading semantic code analysis engine,
CodeQL, against a repository's source code to find security vulnerabilities.

https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning-with-codeql

https://github.com/ossf/scorecard/blob/main/docs/checks.md#sast
Signed-off-by: naveen <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

**Comment by Uzlopak** [comment]
Well, vkarpov is not fan of automatic code analysis i think. Also i dont know if we should run it on every pull request. It takes on a test project of mine like 7 minutes... For 3 lines of code. Also we configured it properly on lgtm so you should be able to get codeql data by visiting lgtm.com.

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        # Override language selection by uncommenting this and choosing your languages
+        # with:
+        #   languages: go, javascript, csharp, python, cpp, java
+
+      # Autobuild attempts to build any compiled languages (C/C++, C#, or Java).
+      # If this step fails, then you should remove it and run the build manually (see below).
+      - name: Autobuild
```
Comment (Uzlopak) [inline .github/workflows/codeql.yml:28]: Autobuild is not necessary for javascript

Comment (naveensrinivasan) [reply .github/workflows/codeql.yml]: Took care of it. Thanks

**Comment by Uzlopak** [comment]
Create a codeql configuration yml and restrict the codeql to lib folder.

**Comment by Uzlopak** [comment]
```
name: "My CodeQL config"

languages: 
  - 'javascript'
paths:
  - lib
```

**Comment by naveensrinivasan** [comment]
> Create a codeql configuration yml and restrict the codeql to lib folder.

👍

**Comment by Uzlopak** [comment]
`1. issue was detected with this workflow: Please specify an on.push hook so that Code Scanning can compare pull requests against the state of the base branch.`

**Comment by naveensrinivasan** [comment]
> `1. issue was detected with this workflow: Please specify an on.push hook so that Code Scanning can compare pull requests against the state of the base branch.`

Updated with on push

**Review by vkarpov15 — APPROVED** [review]
@Uzlopak is right, I normally don't like automated scanning tools because in my experience they're just glorified linters. The ones I've worked with in the past claim to help you "find bugs" but can't really do anything more sophisticated than tell you to sort your imports and use `===` instead of `==` in my experience.

I'm impressed with what CodeQL has managed to find though. Detecting prototype pollution is :+1: . I'll merge this one, we'll see how this tool works.

**Comment by Uzlopak** [comment]
Ok, then just modify the triggering events to just run codeql if we have significant code changes.

I suggest:

```
  pull_request:
    paths:
      - 'lib/**'
  push:
    branches:
      - master
    paths:
      - 'lib/**'
```

Then we should be good to go. :)

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: "Code Scanning - Action"
+
+on:
+  pull_request:
```
Comment (Uzlopak) [inline .github/workflows/codeql.yml:4]:
```suggestion
  pull_request:
    paths:
      - 'lib/**'
```

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  push:
```
Comment (Uzlopak) [inline .github/workflows/codeql.yml:5]:
```suggestion
  push:
    branches:
      - master
    paths:
      - 'lib/**'
```

**Comment by Uzlopak** [comment]
Thank you :)

**Pull request merged**